src/backend/columnar/columnar_customscan.c

```diff
@@ -825,8 +1273,40 @@ ColumnarScan_ExplainCustomScan(CustomScanState *node, List *ancestors,
 
 	if (scanDesc != NULL)
 	{
-		int64 chunkGroupsFiltered = ColumnarScanChunkGroupsFiltered(scanDesc);
-		ExplainPropertyInteger("Columnar Chunk Groups Removed by Filter", NULL,
-							   chunkGroupsFiltered, es);
+		CustomScan *cscan = castNode(CustomScan, node->ss.ps.plan);
+		List *chunkGroupFilter = cscan->custom_exprs;
+
+		if (chunkGroupFilter != NIL)
+		{
+			Expr *conjunction;
+
+			if (list_length(chunkGroupFilter) == 1)
+			{
+				conjunction = (Expr *) linitial(chunkGroupFilter);
+			}
+			else
+			{
+				conjunction = make_andclause(chunkGroupFilter);
+			}
+
+#if PG_VERSION_NUM >= 130000
+			List *context = set_deparse_context_plan(es->deparse_cxt,
+													 &cscan->scan.plan,
+													 ancestors);
+#else
+			PlanState *planstate = &node->ss.ps;
+			List *context = set_deparse_context_planstate(es->deparse_cxt,
+														  (Node *) planstate,
+														  ancestors);
+#endif
+
+			char *str = deparse_expression((Node *) conjunction,
+										   context, false, false);
+			ExplainPropertyText("Columnar Chunk Group Filters", str, es);
+
+			ExplainPropertyInteger(
+				"Columnar Chunk Groups Removed by Filter",
+				NULL, ColumnarScanChunkGroupsFiltered(scanDesc), es);
```

Comment:
what about putting this whole block into a function like `ExplainChunkGroupFiltersIfExists` ?